analyses/snv-callers/vaf_cutoff_experiment.Rmd

```diff
@@ -0,0 +1,285 @@
+Regardless, VarDict's data should still be dropped as it is too aberrant from the other callers as 
+well as overly sensitive. 
+
+#### Usage
+
+To run this from the command line, use:
+```
+Rscript -e "rmarkdown::render('analyses/snv-callers/vaf_cutoff_experiment.Rmd', 
+                              clean = TRUE)"
+```
+
+_This assumes you are in the top directory of the repository._
+
+## Setup
+
+#### Packages and functions
+
+Read in set up script.
+
+```{r}
+if (!("GGally" %in% installed.packages())) {
+  install.packages("GGally")
+}
+if (!("ggupset" %in% installed.packages())) {
+  install.packages("ggupset")
+}
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+```
+
+Set up output directories. 
+
+```{r}
+base_results_dir <- "results"
+base_plots_dir <- "plots"
+```
+
+Make new directories for the comparison analysis.
+
+```{r}
+vaf_cutoff_results_dir <- file.path(base_results_dir, "vaf_cutoff_data")
+
+# Make caller specific plots folder
+if (!dir.exists(vaf_cutoff_results_dir)) {
+  dir.create(vaf_cutoff_results_dir)
+}
+```
+
+Function for creating a TMB correlation matrix amongst the callers for a particular cutoff.
+
+```{r}
+cor_matrix_plot <- function(cutoff) {
+  # For a given VAF cutoff, calculate TMB correlations amongst the callers and plot it using
+  # GGally::ggpairs
+  #
+  # Args:
+  #   cutoff: The VAF cutoff set to analyze
+  #
+  # Returns:
+  #   A correlation matrix plot amongst the callers
+
+  # Isolate the TMB stats for this cutoff
+  cutoff_df <- tmb_filter_long %>%
+    dplyr::filter(vaf_cutoff == cutoff) %>%
+    dplyr::select("lancet", "mutect2", "strelka2", "vardict")
+
+  # Make the plot
+  cor_plot <- GGally::ggpairs(cutoff_df, mapping = ggplot2::aes(alpha = 0.05)) +
+    ggplot2::theme_classic() +
+    ggplot2::ggtitle(paste0("VAF cutoff: ", cutoff))
+
+  # Return the plot
+  return(cor_plot)
+}
+```
+
+Function for correlating TMB statistics for each VAF cutoff set. 
+
+```{r}
+cor_by_vaf <- function(cutoff = 0.1) {
+  # For a given VAF cutoff, isolate the TMB statistics and correlate them.
+  #
+  # Args:
+  #   cutoff: the VAF cutoff to calculate the correlation by.
+  #
+  # Returns:
+  #   R values for Spearman's correlation amongst the callers
+
+  cutoff_df <- tmb_filter_long %>%
+    dplyr::filter(vaf_cutoff == cutoff) %>%
+    dplyr::select(-Tumor_Sample_Barcode, -vaf_cutoff)
+
+  # Do correlations between callers
+  cor_res <- cor(cutoff_df, method = "spearman", use = "na.or.complete")
+
+  # This is to remove redundancy as upper correlation matrix == lower
+  cor_res[upper.tri(cor_res, diag = TRUE)] <- NA
+
+  # Format into long format data.frame
+  cor_df <- reshape2::melt(cor_res, na.rm = TRUE, value.name = "cor") %>%
+    dplyr::mutate(
+      compare = paste0(Var1, "-", Var2),
+      cutoff
+    )
+
+  # Return a data.frame ready for being added to the bigger data.frame for plotting.
+  return(cor_df)
+}
+```
+
+Calculate variance for a sample for a given caller amongst the different VAF cutoffs.
```

Comment:
I don't understand what this is calculating. Variance of what? (Okay, I figured it out: Variance of TMB, but you should state that here. Also, I tend to prefer sd, as it tends to fit intuitions better.